modules/core/src/smithy4s/syntax/package.scala

```diff
@@ -113,4 +113,10 @@ package object syntax
       OneOf(oneOf.label, oneOf.schema.withHints(hints: _*), oneOf.inject)
   }
 
+  implicit class withMaskSyntax[F[_]](val s: Schematic[F]) extends AnyVal {
+    def mask(hintMask: HintMask): Schematic[F] = HintMask.mask(s, hintMask)
+  }
+
+  implicit def newTypeToHintKey[A](a: Newtype[A]): Hints.Key[_] = a.key
```

Comment:
What happens if you return `Hints.Key[A]` here ? Does the generated code still compile ? If it does, we should do that, and maybe move this implicit def into the companion object of Hints.Key

Reply:
Cool, I will give that a try

Reply:
Doesn't compile like that:
```
type mismatch;
 found   : smithy4s.Hints.Key[a.Type]
 required: smithy4s.Hints.Key[A]
```

Reply:
Moved to the companion object though as I think that is better